eZ/Publish/Core/settings/storage_engines/legacy/search_query_handlers.yml

```diff
@@ -303,6 +304,12 @@ services:
         tags:
             - {name: ezpublish.persistence.legacy.search.gateway.criterion_handler.content}
 
+    ezpublish.persistence.legacy.search.gateway.criterion_handler.content.permission_subtree:
```

Comment:
I agree, if we can rely on container respecting the order of service definition, that would be the best approach, as we can then keep this knowledge outside Solr engine.